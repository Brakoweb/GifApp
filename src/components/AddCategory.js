import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [category, setCategory] = useState('');
    
    const handleOnChange = ({target}) => {
        setCategory(target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        
        if(category.trim().length > 2){
            setCategories( categories => [...categories, category]);
            setCategory('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type = "text"
                value = {category}
                placeholder = 'Escribe una Categoria'
                onChange = {handleOnChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


