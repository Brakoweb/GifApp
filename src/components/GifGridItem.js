import React from 'react';

export const GifGridItem = ( {title, url} ) => {
    return (
        <li>
            <img src={url} alt={title} />
            <p>{title}</p>
        </li>
    )
}
