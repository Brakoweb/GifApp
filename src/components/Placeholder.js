import React from 'react'

export const Placeholder = ({categories}) => {
    console.log(categories.length)
    return (
        <div>
            {categories.length < 1 && <p className="placeholder"> Please write a category.</p> }
        </div>
    )
}
