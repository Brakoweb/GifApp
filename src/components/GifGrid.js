import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

const GifGrid = ({category}) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs(category).then(setImages)
    }, [category])

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {images.map(img => (
                            <GifGridItem 
                                key = {img.id}
                                {...img}
                            />
                    )
                )}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}

export default GifGrid;
