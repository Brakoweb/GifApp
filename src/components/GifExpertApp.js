import React, {useState} from 'react'
import { AddCategory } from './AddCategory';
import GifGrid from './GifGrid';
import { Placeholder } from './Placeholder';

const GifExpertApp = () => {
    const [categories, setCategories] = useState([]);

    const handleReset = () => setCategories([]);

    return (
        <div>
            <h2>GifApp</h2>
            <AddCategory setCategories = {setCategories} />
            <hr />
            <ol>
                {categories.map( element => 
                    <GifGrid category={element} key={element} />
                )}
            </ol>

            <Placeholder categories={categories} />

            <button className="go-top" onClick={handleReset}>Reset</button>
        </div>
    )
}

export default GifExpertApp;
