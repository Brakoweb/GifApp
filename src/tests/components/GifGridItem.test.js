import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Test in <GifGridItem />', () => {
    const title = 'Title';
    const url = 'https://url.com/';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('must show the component correctly', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('must have a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('should have the image equal to url and alt of the props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('should have animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
    });
    
    
})
