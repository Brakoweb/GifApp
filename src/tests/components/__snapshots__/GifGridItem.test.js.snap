// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> must show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="https://url.com/"
  />
  <p>
    Title
  </p>
</div>
`;
